src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -116,7 +117,7 @@ export class DeviceDetails extends Component {
               <Accordion title={t('devices.details.tags.title')} description={t('devices.details.tags.description')}>
                 <Grid>
                   {
-                    (device.tags && device.tags.length) &&
+                    (device.tags && Object.entries(device.tags).length > 0) &&
```

Comment:
Use `Object.keys` instead, that is the convention for object size and you don't need the values here anyway.